strawberry/types/_mapped_scalar.py

```diff
@@ -0,0 +1,7 @@
+class _StrawberryMappedScalar:
+    def __init__(self, scalar_type: type) -> None:
+        self.scalar_type = scalar_type
+
+
+def _mapped_scalar(cls: type) -> type:
+    return _StrawberryMappedScalar(cls)
```

Comment:
The way I implemented this PR right doesn't really use this, so maybe we can just get rid of it? 🤔 

The idea was to do something like `GlobalID = Annotated[_GlobalID, _mapped_scalar(ID)]`, but we are not using that annotated value at all 😊